### ISSUE TITLE
Install using proxy does not work on v0.4.0-alpha.5

### ISSUE THREAD

**Issue by salkin** [talos-systems/talos#1901]
When trying to use a init.yaml for install using proxy there is no proxy used on the image pull for installer. 

Boot logs:
``` 
[    8.051058] [talos] [phase]: start system-containerd
[    8.054725] [talos] [phase]: start system-containerd done, 3.618312ms
[    8.058358] [talos] [phase]: platform tasks
[    8.061417] [talos] service[system-containerd](Preparing): Running pre state
[    8.064146] [talos] service[system-containerd](Preparing): Creating service runner
[    9.300229] [talos] service[system-containerd](Running): Process Process(["/bin/containerd" "--address" "/run/system/containerd/containerd.sock" "--state" "/run/system/containerd" "--root" "/run/system/lib/containerd"]) started with PID 1694
[    9.381357] [talos] pulling installer container image: "docker.io/autonomy/installer:latest"
[    9.383969] [talos] service[system-containerd](Running): Health check successful
[   69.389350] [talos] [phase]: platform tasks error running task: 2 error(s) occurred:
[   69.393620]  failed to pull image "docker.io/autonomy/installer:latest": failed to resolve reference "docker.io/autonomy/installer:latest": failed to do request: Head https://registry-1.docker.io/v2/autonomy/installer/manifests/latest: dial tcp 54.210.105.17:443: i/o timeout
[   69.403866]  timeout
[   69.404999] [talos] [phase]: platform tasks done, 1m1.346641151s
[   69.408573] [talos] error running phase "platform tasks": 1 error occurred:
[   69.411423]  * 2 error(s) occurred:
[   69.412683]  failed to pull image "docker.io/autonomy/installer:latest": failed to resolve reference "docker.io/autonomy/installer:latest": failed to do request: Head https://registry-1.docker.io/v2/autonomy/installer/manifests/latest: dial tcp 54.210.105.17:443: i/o timeout
[   69.419425]  timeout
[   69.420196]
[   69.420746]
[   69.421256] [talos] recovered from: boot failed: error running phase "platform tasks": 1 error occurred:
[   69.423133]  * 2 error(s) occurred:
[   69.423923]  failed to pull image "docker.io/autonomy/installer:latest": failed to resolve reference "docker.io/autonomy/installer:latest": failed to do request: Head https://registry-1.docker.io/v2/autonomy/installer/manifests/latest: dial tcp 54.210.105.17:443: i/o timeout
[   69.428547]  timeout
[   69.429001]
[   69.429357]
[   69.429720] [talos] rebooting in 10 seconds
[   70.430615] [talos] rebooting in 9 seconds
[   71.432935] [talos] rebooting in 8 seconds
```

init.yaml:
``` 
version: v1alpha1
machine:
  env:
    http_proxy: http://10.144.1.10:8080
    https_proxy: http://10.144.1.10:8080
  type: init
...
```

When using initramfs v0.4.0-alpha-4 the install is successful but fails naturally later when it tries to proxy the unix calls.

**Comment by smira** [comment]
@salkin this might be related to the changes for registry mirrors, it doesn't pick up proxy settings from the config, I'll get a fix today

**Comment by salkin** [comment]
Applying your commit solved that issue of install but after the install to disk and on the next boot when it tries to pull etcd and hyperkube it still does not try to use the proxy.

```
064405b73172863781153fcf4aec4b)
[   20.620753] [talos] service[ntpd](Preparing): Creating service runner
[   20.676701] [talos] service[ntpd](Running): Started task ntpd (PID 2268) for container ntpd
[   79.256236] [talos] service[kubelet](Failed): Failed to run pre stage: 2 error(s) occurred:
[   79.260062] 	failed to pull image "k8s.gcr.io/hyperkube:v1.17.1": failed to resolve reference "k8s.gcr.io/hyperkube:v1.17.1": failed to do request: Head https://k8s.gcr.io/v2/hyperkube/manifests/v1.17.1: dial tcp 108.177.15.82:443: i/o timeout
[   79.269292] 	timeout
[   80.520219] [talos] service[etcd](Failed): Failed to run pre stage: failed to pull image "k8s.gcr.io/etcd:3.3.15-0": 2 error(s) occurred:
[   80.523039] 	failed to pull image "k8s.gcr.io/etcd:3.3.15-0": failed to resolve reference "k8s.gcr.io/etcd:3.3.15-0": failed to do request: Head https://k8s.gcr.io/v2/etcd/manifests/3.3.15-0: dial tcp 108.177.15.82:443: i/o timeout
[   80.527676] 	timeout
```

**Comment by smira** [comment]
I can look into that, even though seems strange as it's supposed to be using exactly the same code path.

**Comment by smira** [comment]
@salkin I guess this is the `installer` image, it installed "previous" version of Talos without the fix from #1902.

You can try `docker.io/autonomy/installer:v0.4.0-alpha.5-17-gcafd33ac` image which contains the fix

**Comment by salkin** [comment]
Verified.
